src/datasources.js

```diff
@@ -300,7 +307,14 @@ define([
         }
         function getNodes(source, path, info) {
             var nodes = _.chain(source && source.structure)
-                .map(node(source, path, info))
+                .map(function (item, id) {
+                    if (item.merge) {
```

Comment:
Ah. I see. so merge is for the UI layer, not the data layer? 

slightly related: May be useful to have better documentation for this builder function? I find myself having to really dig into the details whenver i look at it